Inferences/InductionHelper.cpp

```diff
@@ -209,22 +208,61 @@ bool InductionHelper::isInductionTermFunctor(unsigned f) {
          );
 }
 
+bool containsSkolem(Term* t) {
+  if (env.signature->getFunction(t->functor())->skolem()) return true; 
```

Comment:
What about this?
Since induction is by default only applied to simple terms, let's have a special case for that (so we avoid creating the iterator). Otherwise (if t is complex) we use NonVariableIterator.

Reply:
Ahah, OK - I see the logic now! My mistake. The original is probably fine then, but it looks like this is a further improvement.

Just while we're here, could the two helper functions `containsSkolem` and `termAndLiteralSatisfyStrictness` be `static`?

Reply:
Done.